planet/clients/subscriptions.py

```diff
@@ -59,15 +99,14 @@ async def list_subscriptions(self,
 
         """
         try:
-            select_subs = (dict(**sub, id=sub_id) for sub_id,
-                           sub in _fake_subs.items()
-                           if not status or sub['status'] in status)
-            filtered_subs = itertools.islice(select_subs, limit)
-
-            for sub in filtered_subs:
+            # TODO: replace with httpx request.
+            async for sub in _server_subscriptions_get(status=status,
+                                                       limit=limit):
                 yield sub
-
         except Exception as server_error:
+            # TODO: remove "from server_error" clause. It's useful
+            # during development but may invite users to depend on the
+            # value of server_error.
```

Comment:
@jreiberkyle I thought hard about this today and asked around and now I feel chained exceptions create too much "API surface" that users might attach onto. I'm completely aligned with not raising chained exceptions in the final release.

Reply:
this is great to hear. thank you for the consideration on this point.